### PR TITLE
stats: refine estimate in buckets

### PR DESCRIPTION
**Pull request by lamxTyler**
Previously, when the query range lies in the bucket, we use half of the bucket depth as an estimation. This PR uses the continuous-value assumption to estimate it.

PTAL @coocood  @hanfei1991  @winoros

### CODE REVIEW

File: statistics/scalar.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		return 1
+	}
+	frac := (valueScalar - lowerScalar) / (upperScalar - lowerScalar)
+	if math.IsNaN(frac) || math.IsInf(frac, 0) || frac < 0 || frac > 1 {
+		return 0.5
+	}
+	return frac
+}
+
+// convertToScalar will convert the datum to scalar values.
```
Comment (shenli) [inline statistics/scalar.go]: will convert -> converts

File: statistics/scalar.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		if lower[common] == upper[common] && lower[common] == value[common] {
+			common++
+		} else {
+			break
+		}
+	}
+	return convertOneBytesToScalar(lower[common:]), convertOneBytesToScalar(upper[common:]), convertOneBytesToScalar(value[common:])
+}
+
+func convertOneBytesToScalar(value []byte) float64 {
```
Comment (coocood) [inline statistics/scalar.go]: This function is complex, how about using `BigEndian.Uint64`?
```
var buf [8]byte
copy(buf[:], value)
return float64(BigEndian.Uint64(buf[:])
```

**Comment by coocood** [comment]
LGTM

**Pull request merged**